pynamodb/tests/test_attributes.py

```diff
@@ -723,6 +737,7 @@ def test_list_of_map_with_of(self):
         class Person(MapAttribute):
             name = UnicodeAttribute()
             age = NumberAttribute()
+            extra_arbitrary_data = JSONAttribute()
```

Comment:
Yes, this would failed before the PR: when the Person is serialized, the extra_arbitrary_data  attribute was mapped to a MapAttribute instead of JSONAttribute because _get_class_for_serialize() returns MapAttribute class for a dict.